packages/test/src/test-sinks.ts

```diff
@@ -118,6 +118,7 @@ if (RUN_INTEGRATION_TESTS) {
       memo: {},
       parent: undefined,
       searchAttributes: {},
+      typedSearchAttributes: JSON.parse(JSON.stringify(new TypedSearchAttributes())),
```

Comment:
Looks like we might have more and more discrepencies between the actual `WorkflowInfo` interface and that sink functions see of it. We already had the `unsafe.now()`, mentionned below, and now, `typedSearchAttributes` will be plain objects/arrays rather than an instance of the `TypedSearchAttributes` class.

We could consider rehydrating the `WorkflowInfo` objects before actually passing them to sink functions, but I honestly don't think there's a strong use case for it enough to justify the extra CPU cycle and memory churning that would involve. So I think we may simply create a variant of the `WorkflowInfo` interface that corresponds to what we actually get in sinks.

Don't fix this just now, but please add a FIXME and/or open a ticket for that.

Reply:
Yup, just created a literal to satisfy the test and added a FIXME as discussed offline.